pyeasyga/pyeasyga.py

```diff
@@ -135,13 +136,34 @@ def create_initial_population(self):
             initial_population.append(individual)
         self.current_generation = initial_population
 
-    def calculate_population_fitness(self):
+    def calculate_population_fitness(self, n_workers=None, parallel_type="processing"):
         """Calculate the fitness of every member of the given population using
         the supplied fitness_function.
         """
-        for individual in self.current_generation:
-            individual.fitness = self.fitness_function(
-                individual.genes, self.seed_data)
+        # If using a single worker, run on a simple for loop to avoid losing
+        # time creating processes.
+        if n_workers == 1:
+
+            for individual in self.current_generation:
+                individual.fitness = self.fitness_function(
+                    individual.genes, self.seed_data)
+        else:
+
+            if "process" in parallel_type.lower():
+                executor = futures.ProcessPoolExecutor(max_workers=n_workers)
+            else:
+                executor = futures.ThreadPoolExecutor(max_workers=n_workers)
+
+            # Create two lists from the same size to be passed as args to the
+            # map function.
+            genes = [individual.genes for individual in self.current_generation]
+            data = [self.seed_data for _ in self.current_generation]
```

Comment:
`data = [self.seed_data] * len(self.current_generation)` should be noticeably faster given large population and seed_data sizes.